docs/cas-server-documentation/ux/User-Interface-Customization-Views.md

```diff
@@ -30,6 +29,12 @@ Upon a successful `200` status result, the response body is expected to contain
  
 To see the relevant list of CAS properties, please [review this guide](../configuration/Configuration-Properties.html#views).
 
+## Thymeleaf
+
+CAS uses [Thymeleaf (v3.x)](https://www.thymeleaf.org) for its markup rendering engine. Each template is decorated by `layout.html` template file, which provides a layout structure for the template's content. Individual components optimized for re-use among multiple templates are stored in the `src/main/resources/templates/fragments` folder, and referenced by the templates in `src/main/resources/templates`.
```

Comment:
Avoid using version numbers. There is no guarantee that link points to v3, or that it will 2 weeks from now.